recipe/meta.yaml

```
@@ -19,16 +19,17 @@ build:
 requirements:
   host:
     - pip
-    - python >=3.8
+    - python >=3.9
```

As we specify [Python 3.8+ in `setup.py`]( https://github.com/dask/dask-image/blob/v2023.08.0/setup.py#L64 ) should this match?

```suggestion
    - python >=3.8
```

---

Good catch!

Since python 3.8 is officially dropped (https://github.com/dask/dask-image/pull/315), probably this should be corrected in `setup.py`?

---

Maybe we should do a patch release then? The `whl`s and `sdist` will be installable on Python 3.8 given this constraint

---

With patch release do you mean a 2023.08.1 right after this one? That'd make sense to me.

(This comment had got stuck behind the review submission earlier)

---

Submitted PR ( https://github.com/dask/dask-image/pull/336 ) to bump that version

Since the `whl`s and `sdist` are allowing Python 3.8, would it make sense to stick with Python 3.8 one more release?

Alternatively we can do a patch release with the PR above

Think it would be good if the packages are installed in a consistent manner across distributions (as we saw today with Dask & Arrow inconsistency can lead to surprising behavior for users)

---

> Since the whls and sdist are allowing Python 3.8, would it make sense to stick with Python 3.8 one more release?

https://github.com/dask/dask-image/pull/315 dropped 3.8 not because of a specific incompatibility, but the NEP29 recommendation to drop support for python 3.8 after 04/2023. So in principle supporting 3.8 should be okay. 

Inconsistencies/disadvantages that come to mind would be:
- testing was done for 3.9+
- the release notes mention dropping 3.8 as a highlight 😅
- `setup.py` has 3.9+ classifiers

---

Yeah if we would rather do a patch release with PR ( https://github.com/dask/dask-image/pull/336 ) and just bump this PR to 2023.8.1, that seems reasonable to me

---

Oh I see. Just to clarify that I understood you correctly. In terms of concrete steps, your suggestion would be to:
1) Create a 2023.8.1 release on github.
2) Upload to pypi
3) Bump this PR by modifying
  - `set version = "2023.8.0"` to `set version = "2023.8.1"`
  - `set sha256 = "f68064320858ca687d05201277fb9937f47ff0cfc97255689ed8ad558a2a40b2"` to ...?

This would lead to 2023.08.0 being available only on pypi and not conda.

---

Yeah we can also yank 2023.8.0 on PyPI so it wouldn't show up on either

---

Ah yes. I'm thinking what that'd mean for the release notes. We could probably just include the same notes for 2023.08.0 with a comment saying they're shown again since 2023.08.1 is a patch release.

---

How about deleting the v2023.08.0
- github release
- git tag and
- pypi release files

and release it under the same version again (including https://github.com/dask/dask-image/pull/336). That's probably a bad option?

---

Yeah users may already have the release now in some form. Removal could cause confusion

Adding a new release is typically the preferred path

---

FWIW here's how one project [updates release notes for the kinds of issues]( https://github.com/nicolargo/glances/blob/407a3cde210013faca35af0ff7bd652787d8f86f/NEWS.rst#version-3403 )

---

I see, that makes sense! Guess I was wondering about the release notes for 2023.08.0 missing on pypi if we yank it, but people could just refer to the complete version list on github.

Thank you very much John for all your efforts here and patience with me :)

I'll need to get some sleep now and tomorrow morning will create the patch release and work on this PR as discussed.

---

Of course! Happy to help 🙂

Sorry this has been a bit of a bumpy ride here

Hopefully we can learn from these issues and make the experience smoother in the future

---

I think this is a good plan.